### PR TITLE
[aot] Support texture in Metal

### PR DESCRIPTION
**Pull request by PENGUINLIONG**
Issue: #

### CODE REVIEW

File: taichi/rhi/metal/metal_device.mm [hunk 1]
```diff
@@ -179,6 +205,38 @@
...
+                              ImageSamplerConfig sampler_config) {
+  RHI_ASSERT(alloc.device == (Device *)device_);
+  const MetalImage &image = device_->get_image(alloc.alloc_id);
+
+  MetalShaderResource rsc{};
+  rsc.ty = MetalShaderResourceType::texture;
+  rsc.binding = binding;
+  rsc.texture.texture = image.mtl_texture();
+  rsc.texture.is_sampled = true;
+  resources_.emplace_back(std::move(rsc));
```
Comment (bobcao3) [inline taichi/rhi/metal/metal_device.mm:219]: nit: use `push_back(std::move())`

File: taichi/rhi/metal/metal_device.mm [hunk 2]
```diff
@@ -179,6 +205,38 @@
...
+                                                    int lod) {
+  RHI_ASSERT(alloc.device == (Device *)device_);
+  const MetalImage &image = device_->get_image(alloc.alloc_id);
+
+  MetalShaderResource rsc{};
+  rsc.ty = MetalShaderResourceType::texture;
+  rsc.binding = binding;
+  rsc.texture.texture = image.mtl_texture();
+  rsc.texture.is_sampled = false;
+  resources_.emplace_back(std::move(rsc));
```
Comment (bobcao3) [inline taichi/rhi/metal/metal_device.mm:235]: push_back

File: taichi/rhi/metal/metal_device.mm [hunk 3]
```diff
@@ -403,8 +474,29 @@ DeviceCapabilityConfig collect_metal_device_caps(MTLDevice_id mtl_device) {
...
+      [mtl_device newSamplerStateWithDescriptor:desc];
+
+  [desc release];
+
+  return std::make_unique<MetalSampler>(mtl_sampler_state);
+}
+
 MetalDevice::MetalDevice(MTLDevice_id mtl_device) : mtl_device_(mtl_device) {
   compute_stream_ = std::unique_ptr<MetalStream>(MetalStream::create(*this));
+  graphics_stream_ = std::unique_ptr<MetalStream>(MetalStream::create(*this));
```
Comment (bobcao3) [inline taichi/rhi/metal/metal_device.mm:497]: Has the `semaphore` between streams been implemented yet? Without that true-multi-stream will break.

Comment (PENGUINLIONG) [reply taichi/rhi/metal/metal_device.mm:497]: oof, i'll reuse compute_stream then

**Review by bobcao3 — APPROVED** [review]
LGTM!

**Pull request merged**